lib/GenerateResource.js

```diff
@@ -0,0 +1,191 @@
+
+/**
+ * Initialize this repository and read in all of the strings.
+ *
+ * @param {Project} project the current project
+ * @param {Function(Object, Object)} cb callback to call when the
+ * initialization is done
+ */
+GenerateResource.prototype.init = function() {
+    var dir = this.xliffsDir;
+    var list;
+    try {
+        list = fs.readdirSync(dir);
+    } catch (err) {
+        logger.warn(dir + "is invalid directory");
+    }
+    
+    if (list) {
+        list.filter(function(file) {
+            var match = xliffFileFilter.exec(file);
+            if (!match ||
+               match.length < 2 ||
+               (match.length >= 3 && match[2] && !utils.iso639[match[2]]) ||
+               (match.length >= 5 && match[4] && !utils.iso15924[match[4]]) ||
+               (match.length >= 7 && match[6] && !utils.iso3166[match[6]])) {
+                return false;
+            }
+            return true;
+        }).forEach(function (file) {
+            pathName = path.join(dir, file);
+            var xliff = new Xliff({
+                sourceLocale: this.sourceLocale
+            });
+            if (fs.existsSync(pathName)) {
+                var data = fs.readFileSync(pathName, "utf-8");
+                xliff.deserialize(data);
+                var ts = xliff.getTranslationSet();
+                this.ts.addSet(ts);
+            } else {
+                logger.warn("Could not open xliff file: " + pathName);
+            }
+        }.bind(this));
+    }
+
+    if (this.pathName) {
+        var xliff = new Xliff({
+            sourceLocale: this.sourceLocale
+        });
+        if (fs.existsSync(this.pathName)) {
+            var data = fs.readFileSync(this.pathName, "utf-8");
+            xliff.deserialize(data);
+            var ts = xliff.getTranslationSet();
+            this.ts.addSet(ts);
+        } else {
+            logger.warn("Could not open xliff file: " + this.pathName);
+        }
+    }
+    if (this.ts.resources.length) {
+        logger.info("Generating resource size: " + this.ts.resources.length);
+        this.generate();
+    }
+};
+
+GenerateResource.prototype._calResourcePath = function(resLocale) {
+    var resPath = "";
+    var splitLocale = resLocale.split("-");
+    var minimalLocale = new LocaleMatcher({locale: resLocale}).getLikelyLocaleMinimal().getSpec();
+    if (resLocale === minimalLocale ) {
+        resPath = splitLocale[0];
+    } else {
+        resPath = splitLocale.join("/");
+    }
+    return resPath;
+}
+
+GenerateResource.prototype.getResourceFilePath = function(resLocale) {
+    var resPath = this._calResourcePath(resLocale);
+    return path.join(this.targetDir, this.resRoot, resPath, this.resName);
+}
+
+GenerateResource.prototype.writeFiles = function() {
+    var manifest = {
+        files: []
+    }
+    var fullPath;
+    for (locale in this.output){
+        var resourcePath = this._calResourcePath(locale);
```

Comment:
This is hard-coded to write out json files in webos format. Each resource identifies what type it is and what class handles it. This code should get the resource file type for each resource, add the resource to it, and then ask each resource file type to write itself out. That way, `generate` can be used with any project, not just webos projects.

Reply:
Yeah, I removed codes here. then made it to generate the file through plugins. plugin info will be extracted project configuration. the config file is an essential file for localization. so I think it's ok to use that info.
and I've defined newly process to generate localization which omitted generatePseudo, extract and write task.
https://github.com/iLib-js/loctool/blob/createGenerateMode/loctool.js#L466

Reply:
I don't see those changes here... are they pushed to github yet?